src/core/CLucene/index/IndexWriter.cpp

```diff
@@ -1383,7 +1388,7 @@ void IndexWriter::indexCompaction(std::vector<lucene::store::Directory *> &src_d
             proxOutputList.push_back(proxOut);
             // Instantiate a new termInfosWriter which will write in directory
             // for the segment name segment using the new merged fieldInfos
-            TermInfosWriter *termInfosWriter = _CLNEW TermInfosWriter(dest_dir, segment.c_str(), fieldInfos, termIndexInterval);
+            auto* termInfosWriter = _CLNEW STermInfosWriter<char>(dest_dir, segment.c_str(), fieldInfos, termIndexInterval);
```

Comment:
Why we change STermInfosWriter here?

Reply:
Because we have two types in clucene: STermInfosWriter and TermInfosWriter, I only modified the code logic in STermInfosWriter. My expectation is to maintain only one process within STermInfosWriter.